src/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -486,6 +555,53 @@ contract BLSRegistryCoordinatorWithIndices is EIP712, Initializable, IBLSRegistr
         blsPubkeyRegistry.initializeQuorum(quorumNumber);
     }
 
+    /**
+     * @notice Record an update to an operator's quorum bitmap.
+     * @param newBitmap is the most up-to-date set of bitmaps the operator is registered for
+     */
+    function _updateOperatorBitmap(bytes32 operatorId, uint192 newBitmap) internal {
+
+        uint256 historyLength = _operatorBitmapHistory[operatorId].length;
+
+        if (historyLength == 0) {
+            // No prior bitmap history - push our first entry
+            _operatorBitmapHistory[operatorId].push(QuorumBitmapUpdate({
+                updateBlockNumber: uint32(block.number),
+                nextUpdateBlockNumber: 0,
+                quorumBitmap: newBitmap
+            }));
+        } else {
+            // We have prior history - fetch our last-recorded update
+            QuorumBitmapUpdate storage lastUpdate = _operatorBitmapHistory[operatorId][historyLength - 1];
+
+            /**
+             * If the last update was made in the current block, update the entry.
+             * Otherwise, push a new entry and update the previous entry's "next" field
+             */
+            if (lastUpdate.updateBlockNumber == uint32(block.number)) {
+                lastUpdate.quorumBitmap = newBitmap;
+            } else {
+                lastUpdate.nextUpdateBlockNumber = uint32(block.number);
+                _operatorBitmapHistory[operatorId].push(QuorumBitmapUpdate({
+                    updateBlockNumber: uint32(block.number),
+                    nextUpdateBlockNumber: 0,
+                    quorumBitmap: newBitmap
+                }));
+            }
+        }
+    }
```

Comment:
one somewhat subtle change in behavior here:
when an operator totally deregisters, previously their last bitmap entry would look like:
```
updateBlockNumber: X
nextUpdateBlockNumber: deregistrationBlock
quorumBitmap: lastNonzeroBitmap
```
but now we will push another entry that looks like:
```
updateBlockNumber: deregistrationBlock
nextUpdateBlockNumber: 0
quorumBitmap: 0
```
I believe the main "property" change here is that before we (should have) never pushed an update with an empty bitmap, and now we definitely will be.
Seems fine + I believe all behavior is consistent.

Reply:
Yes, exactly - I believe that's in the changelog! The motivation was partially to be consistent with the "latest historical entry" as the operator's current status, because previously it wasn't.

Also this introduces some nice invariants around OperatorStatus that we can write tests for - namely, that if an operator is REGISTERED, they have a nonzero bitmap, and if they're NEVER_REGISTERED or DEREGISTERED, they have a zero bitmap.